src/djimatrice_drone.py

```diff
@@ -0,0 +1,228 @@
+        return result
+    
+    def resume_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result.get("success", False)
+
+    # TODO Implement
+    def land_drone(self):
+        try:
+            print("Attempting to call drone specific service...")
+            #TODO change to actual service call and type
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/drone_task_control', 'dji_sdk/DroneTaskControl')
+            # TODO check service type on drone aka check if 6 is correct
+            request = roslibpy.ServiceRequest({"task": DroneTaskControl.LAND})
+
+            print('Calling land_drone service...')
+            #TODO parse service.call(request)
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Drone landing failed"}
+        # self.land_drone_callback(result)
+        return result
+    
+    def land_drone_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result.get("success", False)
+
+    # TODO Implement
+    def fly_home(self):
+        try:
+            print("Attempting to call drone specific service...")
+            #TODO change to actual service call and type
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/drone_task_control', 'dji_sdk/DroneTaskControl')
+            # TODO check service type on drone aka check if 1 is correct
+            request = roslibpy.ServiceRequest({"task": DroneTaskControl.GO_HOME})
+
+            print('Calling fly_home service...')
+            #TODO parse service.call(request)
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Drone flying home failed"}
+        return result
+
+    def fly_home_drone_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result.get("success", False)
+
+    #TODO Implement
+    def update_mission_helper(self, action):
+        if action == Drone.UpdateMissionAction.CONTINUE_MISSION:
+            # Call corresponding service and return result
+            result = {"success":True, "message":"Vacuously true for testing"}
+        elif action == Drone.UpdateMissionAction.UPDATE_CURRENT_MISSION:
+            # Call corresponding service and return result
+            result = {"success":True, "message":"Vacuously true for testing"}
+        elif action == Drone.UpdateMissionAction.END_AND_HOVER:
+            # Call corresponding service and return result
+            result = {"success":True, "message":"Vacuously true for testing"}
+        return result
+    
+    # TODO: Need to implement update_mission_helper to work
+    def update_mission(self):
```

Comment:
whats the difference between update_mission and upload_mission?

Reply:
If there is a difference, should we add this to the API? (ie both drone.py and operator.py)?

Reply:
update_mission was meant to resolve any intermediate states for the drone after actions are called, such as if a mission ends early for some reason. It could be redundant if all of the actions are thorough though.

Reply:
ok awesome thanks.